src/Instrumentation/Slim/src/SlimInstrumentation.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Slim;
+
+use OpenTelemetry\API\Common\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Common\Instrumentation\Globals;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanInterface;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Slim\App;
+use Slim\Interfaces\InvocationStrategyInterface;
+use Slim\Interfaces\RouteInterface;
+use Slim\Middleware\RoutingMiddleware;
+use Slim\Routing\RouteContext;
+use Throwable;
+
+class SlimInstrumentation
+{
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation('io.opentelemetry.contrib.php.slim');
+
+        /**
+         * Create a span representing the http transaction for Slim\App::handle
+         * @psalm-suppress ArgumentTypeCoercion
+         */
+        hook(
```

Comment:
Actually, I don't understand need for this hook as we already hooking into `RequestHandlerInterface::handle`

Reply:
You're right, it's redundant now. I left it there because I wasn't sure which way I'd end up going, but if having Psr15 instrumentation work out when to create a root span in `RequestHandlerInterface::handle`, then this hook can be removed.